src/schema-validator.ts

```diff
@@ -123,24 +75,24 @@ export const schemaValidator = (log: Logger) => {
     errors?.map(error => (error ? `${error.instancePath} ${error?.message}` : ''))?.filter(error => error !== '') ?? []
 
   const validateConfiguration = (
-    configuration: RepositoryConfiguration,
+    parsedDocument: Document.Parsed<ParsedNode>,
     schema: Schema,
-    cstRepresentation: CSTRepresentation,
+    lineCounter: LineCounter,
   ): ConfigurationValidation => {
-    const validateConfiguration = ajv.compile<RepositoryConfiguration>(schema)
+    const validateConfigurationFn = ajv.compile<RepositoryConfiguration>(schema)
```

Comment:
I'd keep the original naming for this functions, since it's obvious it's a function based on the type of the const - WDYT?

Reply:
If it has to be really obvious, the type can also be explicitly added like so: 

```suggestion
    const validateConfiguration: ValidateFunction<RepositoryConfiguration> = ajv.compile<RepositoryConfiguration>(schema)
```

😄

Reply:
It's fine I can revert the change back, actually it is callable but also has attributes so it's kind of not a function 🫤

Reply:
Nice 😄 

Yeah, true that's a bit strange 🤔 They do call it a function [in the docs](https://ajv.js.org/api.html#ajv-compile-schema-object-data-any-boolean-promise-any), though